engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityCache.java

```diff
@@ -0,0 +1,350 @@
+
+    /**
+     * Creates the entity without sending any events. The entity life cycle subscriber will however be informed.
+     */
+    @Override
+    public EntityRef createEntityWithoutLifecycleEvents(Prefab prefab) {
+        if (prefab != null) {
+            List<Component> components = Lists.newArrayList();
+            for (Component component : prefab.iterateComponents()) {
+                components.add(entityManager.getComponentLibrary().copy(component));
+            }
+            components.add(new EntityInfoComponent(prefab, prefab.isPersisted(), prefab.isAlwaysRelevant()));
+
+            return createEntityWithoutLifecycleEvents(components);
+        } else {
+            return createEntityWithoutLifecycleEvents(Collections.<Component>emptyList());
+        }
+    }
+
+    /**
+     * Destroys the entity without sending any events. The entity life cycle subscriber will however be informed.
+     */
+    @Override
+    public void destroyEntityWithoutEvents(EntityRef entity) {
+        if (entity.isActive()) {
+            entityManager.notifyComponentRemovalAndEntityDestruction(entity.getId(), entity);
+            destroy(entity);
+        }
+    }
+
+    @Override
+    public EntityRef createEntityWithId(long id, Iterable<Component> components) {
+        if (!entityManager.registerId(id)) {
+            return EntityRef.NULL;
+        }
+        for (Component c : components) {
+            componentStore.put(id, c);
+        }
+        EntityRef entity = createEntityRef(id);
+        if (entityManager.getEventSystem() != null) {
+            entityManager.getEventSystem().send(entity, OnActivatedComponent.newInstance());
+        }
+        for (Component component: components) {
+            entityManager.notifyComponentAdded(entity, component.getClass());
+        }
+        return entity;
+    }
+
+    public EntityBuilder newBuilder() {
+        return new EntityBuilder(this);
+    }
+
+    @Override
+    public EntityBuilder newBuilder(String prefabName) {
+        if (prefabName != null && !prefabName.isEmpty()) {
+            Prefab prefab = entityManager.getPrefabManager().getPrefab(prefabName);
+            if (prefab == null) {
+                logger.warn("Unable to instantiate unknown prefab: \"{}\"", prefabName);
+                return new EntityBuilder(this);
+            }
+            return newBuilder(prefab);
+        }
+        return newBuilder();
+    }
+
+    @Override
+    public EntityBuilder newBuilder(Prefab prefab) {
+        EntityBuilder builder = new EntityBuilder(this);
+        if (prefab != null) {
+            for (Component component : prefab.iterateComponents()) {
+                builder.addComponent(entityManager.getComponentLibrary().copy(component));
+            }
+            builder.addComponent(new EntityInfoComponent(prefab, prefab.isPersisted(), prefab.isAlwaysRelevant()));
+        }
+        return builder;
+    }
+
+    public Map<Long, BaseEntityRef> getEntityStore() {
```

Comment:
I hadn't considered that; I think an unmodifiable map would be more sensible, and then have a `putEntity` method so that the EntityManager can put entities into the map (from `createEntityRef`). But perhaps that should be changed too: maybe the EntityManager should just call the cache's `createEntityRef`? I think restricting external changes to the internal map would be an improvement; I'll look into the best way of doing it.